modules/plottinggl/src/processors/scatterplotprocessor.cpp

```diff
@@ -95,29 +95,29 @@ ScatterPlotProcessor::ScatterPlotProcessor()
             }
         });
     selectionChangedCallBack_ =
-        scatterPlot_.addSelectionChangedCallback([this](const std::vector<bool>& selected) {
+        scatterPlot_.addSelectionChangedCallback([this](const BitSet& selected) {
             if (brushingPort_.isConnected()) {
-                BitSet selectedIndices;
+                BitSet indices;
                 auto iCol = dataFramePort_.getData()->getIndexColumn();
                 auto& indexCol = iCol->getTypedBuffer()->getRAMRepresentation()->getDataContainer();
-                for (size_t i = 0; i < selected.size(); ++i) {
-                    if (selected[i]) selectedIndices.add(indexCol[i]);
+                for (auto idx : selected) {
+                    indices.add(indexCol[idx]);
                 }
-                brushingPort_.select(selectedIndices);
+                brushingPort_.select(indices);
             } else {
                 invalidate(InvalidationLevel::InvalidOutput);
             }
         });
     filteringChangedCallBack_ =
-        scatterPlot_.addFilteringChangedCallback([this](const std::vector<bool>& filtered) {
+        scatterPlot_.addFilteringChangedCallback([this](const BitSet& filtered) {
             if (brushingPort_.isConnected()) {
```

Comment:
This check is not necessary anymore?

Reply:
Good catch. This would have prevented selection without a brushing port connection.